### ISSUE TITLE
Toegankelijkheidsverbeteringen Q1-2025

### ISSUE THREAD

**Issue by joeribekker** [open-formulieren/open-forms#4622]
**Versie**: 2.6.13

Ruwe eerste inschattingen op basis van rapport (op te vragen bij Joeri).

* [x] 1.3.1 Info en relaties (3 dagen)
* [x] 1.3.5 Identificeer het doel van de input (config?)
* [x] 1.4.11 Contrast van niet-tekstuele content (config? maar wellicht betere default)
~2.5.3 Label in naam (zelfde als 1.3.1?)~
* [x] #4716

- [x] 1.1.1 Niet-tekstuele content (1 uur)
- [x] 4.1.2 Naam, rol, waarde (formio, echt geen idee)
- [x] 4.1.3 Statusberichten (4 uur)
- [x] #4717

* [x] 1.4.3 Contrast (2 uur)
* [x] 1.4.4 Herschalen van tekst (3 dagen)
* [x] 1.4.10 Reflow (3 dagen)
* [x] #4718 

- [x] 3.3.1 Fout identificatie (6 uur)
- [x] 3.3.2 Labels of instructies (config? maar niet zeker, dacht dat we "j" voor jaar gebruikten in NL)
- [x] 3.3.3
- [x] #4719 

* [x] 2.1.1 Modal sluit knop
* [x] 2.4.1 Skiplink
* [x] 2.4.4 Linkdoel
* [x] 2.4.6 Koppen en labels (2 dagen)
* [x] 2.4.7 Focus zichtbaar (4 uur)
* [x] #4720 

Totaal: 13 dagen werk

**Comment by robinmolen** [comment]
The accessibility improvements will be resolved in dedicated issues:

- [ ] #4716
- [ ] #4717
- [ ] #4718
- [ ] #4719 
- [ ] #4720

**Comment by joeribekker** [comment]
@robinmolen can you edit the original ticket and hang them below the relevant bullet?

**Comment by robinmolen** [comment]
Issue 2.5.3 is about the datepicker inputs that currently aren't connected to their labels. I've discussed this issue with Sergei, and we came to the conclusion that this issues cannot be resolved at this time.

The current datepicker (flatpickr) has some accessibility issues that we cannot solve. In the future, if/when we can use the pure React datepicker, this issue will be solved. Unfortunately we can't do anything at this time

**Comment by sergei-maertens** [comment]
Can't estimate without having seen the accessiblity report(s)

**Comment by robinmolen** [comment]
November 26th a new Accessibility test was performed which resulted into a new rapport. Most items in this rapport are 'small' improvements (color contrasts, changing html elements), but there are also a couple 'larger' issues. The list below contains the rapported accessibilty items, which I've tests with the macOS VoiceOver screenreader:

**Improvements that should be relatively simple:**
- [ ] 1.1.1 (non-textual content) The logo in the pdf contains a textual alternative, but cannot be accessed.
- [ ] 1.3.1 (information and relations) The content component allows the use of H1 headers. Because the form titles also use H1 headers, this could result into a 'weird' HTML hierarchy. The rapport advises that we turn these content H1 headers into H2. Probably best to remove the option to use H1 headers, to keep the editor 'simple' and ensure that content options work as expected.
- [ ] 1.3.1 (information and relations) Missing relations between (submission rapport) PDF field labels and values.
- [ ] 1.3.1 (information and relations) When a (submission rapport) PDF field doesn't have a values, we should provide a "No information provided" alternative. (This is to ensure that users understand that a field was left empty)
- [ ] 1.4.3 (color contrast) Error messages have a contrast of 3,7:1. This should be 4,5:1
- [ ] 1.4.11 (non-textual color contrast) Checkbox and radio button borders have a contrast of 2,9:1. This should be 3,0:1
- [ ] 2.1.1 (navigation - keyboard) Tooltip cannot be accessed by screen readers
- [ ] 2.4.2 (navigation - page title) The title in the PDF meta description contains leading and trailing whitespace
- [ ] 3.1.2 (language) Some texts are non translatable. The rapport mentions the `title` of the select component delete button.
- [ ] 4.1.2 (name, role, value) The dropdown variant of the progress menu doesn't contain `aria-expanded` information

All these smaller improvements should be completable in around 3 days

**Improvements that are a bit more complicated:**
- [ ] 2.4.7 (navigation - focus visibility) When the form is displayed on a small screen, or greatly zoomed-in, the progress menu will be shown as a dropdown menu. When you focus any menu items in the dropdown, and repeatable press the tab key, you can place the focus outside the menu. The menu stays expended, resulting in the focused element not being visible.
This might required some major re-writing.
Could take a couple of days
- [ ] 3.3.3 (error suggestion) When a specific input pattern is expected, the error message should provide the expected format or an example.
With examples in the error messages this could be resolved in 1 day.
- [ ] 4.1.2 (name, role, value) The select component contains problems with `name` and `role` on the select element, `aria-selected="true"` on the parent div, missing aria tags
This could be resolved in 2 days
- [ ] 4.1.3 (status messages) The loading spinners have issues with `aria-live` which results into status updates not being announced
This could be resolved in 1 or 2 days

Estimate (including previously done work): 5 weeks
